src/rebuild/srv.c

```diff
@@ -1129,30 +1129,30 @@ rebuild_try_merge_tgts(struct ds_pool *pool, uint32_t map_ver,
 	 * complete.
 	 */
 	d_list_for_each_entry(task, &rebuild_gst.rg_queue_list, dst_list) {
-		if (uuid_compare(task->dst_pool_uuid, pool->sp_uuid) != 0)
+		if (uuid_compare(task->dst_pool_uuid, pool->sp_uuid) != 0) {
 			/* This task isn't for this pool - don't consider it */
+			merge_task = NULL;
```

Comment:
Why reset "merge_task" for different pool? Consider the following sequence:
[T1: pool_1, ver_10, rebuild] [T2: pool_2: ver_12, rebuild]
The if current T3 is ver_14 for rebuild pool_1, then whether T3 can be merged into T1 or not?

Reply:
hmm, it will not merge the task from different pool.  So if the pool is different, merge_task will be reset to NULL, then keep searching the next one in the list.

And if the merge task is NULL, it will return after the loop.

Reply:
T1 is for pool_1, T2 is for pool_2, T3 is for pool_1. My question is that whether we can merge T1 and T3. If not, why?

Reply:
ah, yes, you are right.